In [2]:
import pandas as pd

In [3]:
database = pd.read_csv('melb_data.csv')
database.describe()

Rooms         Price      Distance      Postcode      Bedroom2  \
count  13580.000000  1.358000e+04  13580.000000  13580.000000  13580.000000   
mean       2.937997  1.075684e+06     10.137776   3105.301915      2.914728   
std        0.955748  6.393107e+05      5.868725     90.676964      0.965921   
min        1.000000  8.500000e+04      0.000000   3000.000000      0.000000   
25%        2.000000  6.500000e+05      6.100000   3044.000000      2.000000   
50%        3.000000  9.030000e+05      9.200000   3084.000000      3.000000   
75%        3.000000  1.330000e+06     13.000000   3148.000000      3.000000   
max       10.000000  9.000000e+06     48.100000   3977.000000     20.000000   

           Bathroom           Car       Landsize  BuildingArea    YearBuilt  \
count  13580.000000  13518.000000   13580.000000   7130.000000  8205.000000   
mean       1.534242      1.610075     558.416127    151.967650  1964.684217   
std        0.691712      0.962634    3990.669241    541.014538    37.273762   
min        0.000000      0.000000       0.000000      0.000000  1196.000000   
25%        1.000000      1.000000     177.000000     93.000000  1940.000000   
50%        1.000000      2.000000     440.000000    126.000000  1970.000000   
75%        2.000000      2.000000     651.000000    174.000000  1999.000000   
max        8.000000     10.000000  433014.000000  44515.000000  2018.000000   

          Lattitude    Longtitude  Propertycount  
count  13580.000000  13580.000000   13580.000000  
mean     -37.809203    144.995216    7454.417378  
std        0.079260      0.103916    4378.581772  
min      -38.182550    144.431810     249.000000  
25%      -37.856822    144.929600    4380.000000  
50%      -37.802355    145.000100    6555.000000  
75%      -37.756400    145.058305   10331.000000  
max      -37.408530    145.526350   21650.000000

In [4]:
database.columns
database.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

In [5]:
data_nulls = database.isnull().sum()
data_nulls[0:20]

Suburb             0
Address            0
Rooms              0
Type               0
Price              0
Method             0
SellerG            0
Date               0
Distance           0
Postcode           0
Bedroom2           0
Bathroom           0
Car               62
Landsize           0
BuildingArea    6450
YearBuilt       5375
CouncilArea     1369
Lattitude          0
Longtitude         0
Regionname         0
dtype: int64

In [6]:
print(database.dropna().shape)
data_final1=database.dropna().size

(6196, 21)


In [7]:
print(database.dropna(axis=1).shape)
data_final2=database.dropna(axis=1).size

(13580, 17)


In [8]:
data_original = database.size
porcent_rows_lossed = data_final1/data_original
porcent_columns_lossed = data_final2/data_original

print((1-porcent_rows_lossed)*100)
print((1-porcent_columns_lossed)*100)

54.37407952871871
19.047619047619047


# handling nulls value

In this point we're seeing that there are many data lossing, i can eliminate the columns which are data lossing (Car,Landsize, BuildingArea, YearBuilt, CouncilArea),we would loss 19.05% of data.

Other option is eliminate the rows with one or more data lossing, with this option i'll 54.37% of data

I'm going to eliminate the columns which are have one or more data lossing.



In [9]:
database = database.dropna(axis=1)
database.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 17 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Landsize       13580 non-null  float64
 13  Lattitude      13580 non-null  float64
 14  Longtitude     13580 non-null  float64
 15  Regionname     13580 non-null  object 
 16  Propertycount  13580 non-null  float64
dtypes: float64(9), int64(1), object(7)
memory usage: 1

# Change de data type of dates


In [10]:
database['Date'] = pd.to_datetime(database['Date'],format='%d/%m/%Y')
database['Date'].head()
database.to_csv('Database_1.csv',index=False) #import the clean database to format csv

# Selection of variables

Select the dependent variable and the independent variables

In [17]:
y = database.Price
x = database.drop(columns='Price')
x.info()
y.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Suburb         13580 non-null  object        
 1   Address        13580 non-null  object        
 2   Rooms          13580 non-null  int64         
 3   Type           13580 non-null  object        
 4   Method         13580 non-null  object        
 5   SellerG        13580 non-null  object        
 6   Date           13580 non-null  datetime64[ns]
 7   Distance       13580 non-null  float64       
 8   Postcode       13580 non-null  float64       
 9   Bedroom2       13580 non-null  float64       
 10  Bathroom       13580 non-null  float64       
 11  Landsize       13580 non-null  float64       
 12  Lattitude      13580 non-null  float64       
 13  Longtitude     13580 non-null  float64       
 14  Regionname     13580 non-null  object        
 15  Propertycount  1358

In [19]:
#codificar primero las variables
from sklearn.tree import DecisionTreeRegressor

model_tree_regressor = DecisionTreeRegressor(random_state=3)
model_tree_regressor.fit(x,y)
model_tree_regressor.predict(x)

ValueError: could not convert string to float: 'Abbotsford'

In [ ]:
print